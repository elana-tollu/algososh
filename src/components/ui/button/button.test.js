import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import userEvent from '@testing-library/user-event';
import renderer from 'react-test-renderer';
import { Direction } from '../../../types/direction';
import { Button } from './button';

describe('Button', () => { 
  test('should render with text', () => {
      const tree = renderer
        .create(<Button  text="Надпись на кнопке"/>)
        .toJSON();

      expect(tree).toMatchSnapshot();
  });

  test('should render without text', () => {
    const tree = renderer
      .create(<Button />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('should render sorting ascending', () => {
    const tree = renderer
      .create(<Button sorting={Direction.Ascending}/>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('should render sorting descending', () => {
    const tree = renderer
      .create(<Button sorting={Direction.Descending}/>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('should be disabled', () => {
    const tree = renderer
      .create(<Button  
        text="Надпись на кнопке" 
        disabled={true}
      />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('should show loader', () => {
    const tree = renderer
      .create(<Button  
        text="Надпись на кнопке" 
        isLoader={true}
      />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('should handle click', () => {
    const handleClick = jest.fn();
    render(<Button  
      text="Нажми меня" 
      onClick={handleClick}
    />)
    
    userEvent.click(screen.getByText('Нажми меня'));

    expect(handleClick).toBeCalled();
  });
});