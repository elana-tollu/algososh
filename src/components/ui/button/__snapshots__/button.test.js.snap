// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button disabled 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  >
    Надпись на кнопке
  </p>
</button>
`;

exports[`Button loading 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Button with text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Надпись на кнопке
  </p>
</button>
`;

exports[`Button without text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;
